File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int index = 0;
        sort(nums.begin(),nums.end());
        for(int i = 0;i<nums.size();i++){
            if(nums[i]==nums[i+1]){
                index = nums[i];
                break;
            }
        }
        return index;
    }
};

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int N = nums.length;
        int val = 0;
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i< N; i++){
        if(map.containsKey(nums[i])){
            val = nums[i];
            break;
        }
        else map.put(nums[i], 1);
        }
        return val;
    }
}